code/datums/weakrefs.dm

```diff
@@ -53,18 +53,15 @@
  * will always hard delete.
  */
 /datum/weakref
-	var/reference
+	VAR_PRIVATE/datum/thing
 
 /datum/weakref/New(datum/thing)
-	reference = REF(thing)
+	src.thing = thing
+	RegisterSignal(thing, COMSIG_PARENT_QDELETING, PROC_REF(on_qdeleting))
```

Comment:
how does this impact Unregister/Register signal costs? 
we should prob update documentation to reflect this new sort of behavior too

Reply:
`weakref/New` captures the cost of `RegisterSignal`--both MTPC and total cost dramatically go down

Also I don't see why the documentation needs to be updated when this is a completely transparent implementation detail